frontend/src/Components/OpenContentItemAccordion.tsx

```diff
@@ -0,0 +1,95 @@
+import { OpenContentItem } from '@/common';
+import {
+    ChevronRightIcon,
+    VideoCameraIcon,
+    BookOpenIcon,
+    InformationCircleIcon
+} from '@heroicons/react/24/solid';
+import React, { useState } from 'react';
+import OpenContentCard from './cards/OpenContentCard';
+
+type OpenContentItemMap = Record<string, OpenContentItem[]>;
+
+export default function OpenContentItemAccordion({
+    items
+}: {
+    items: OpenContentItem[];
+}) {
+    const contentMap: OpenContentItemMap = {};
+    items.forEach((item) => {
```

Comment:
nit: you can use `.map` here and collect the items into another collection instead of having to push.

Reply:
I tried the map approach and I found myself ending up using the push function, I know I'm missing something, I figured the best way to not use the push function was just to use filter on the list of items to separate out the lists and add  them to the map. Here is my logic:

```
    const libraries = items.filter((item) => item.content_type === 'library');
    const videos = items.filter((item) => item.content_type === 'video');
    const helpfulLinks = items.filter(
        (item) => item.content_type === 'helpful_links'
    );
    const others = items.filter(
        (item) =>
            item.content_type != 'helpful_links' &&
            item.content_type != 'library' &&
            item.content_type != 'video'
    );
    if (libraries.length > 0) {
        contentMap.Libraries = libraries;
    }
    if (videos.length > 0) {
        contentMap.Videos = videos;
    }
    if (helpfulLinks.length > 0) {
        contentMap['Helpful Links'] = helpfulLinks;
    }
    if (others.length > 0) {
        contentMap['All Others'] = others;
    }
```
Please let me know your thoughts?  Also, I'm wondering on the use of the map() function, were you hinting me to use something other than a map to organize the collections?